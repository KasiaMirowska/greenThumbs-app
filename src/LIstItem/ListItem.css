.list-item {
    background-color: whitesmoke;
}