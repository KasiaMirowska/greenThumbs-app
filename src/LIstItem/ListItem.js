import React from 'react';
import {Link} from 'react-router-dom';
import TokenService from '../Services/token-service';
import config from '../config';


export default function ListItem(props) {
       
        return (
            <li key={props.id}>
                <h2>{props.name}</h2>
                <p>{props.location.address1}</p>
                <p>{props.location.city}</p>
                <p>{props.location.state}</p>
                <p>{props.location.zip_code}</p>
                <p>{props.phone}</p>
                <p>{props.price}</p>
                <img src={props.img} />
                <a href ={`${props.website }`} target="_blank" rel='noopener noreferrer'><h2>Visit</h2></a>
                {props.rating}

                {(!TokenService.hasAuthToken(config.TOKEN_KEY))? <Link to={'/login'}><button>Create ThumbsUP review</button></Link> :
                <Link to={`/review/${props.id}`}><button>Create ThumbsUP review</button></Link>
                }
               
            </li>
        )
    
    
}

