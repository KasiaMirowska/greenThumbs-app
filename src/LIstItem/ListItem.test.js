import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as BR } from 'react-router-dom';
import ListItem from './ListItem';

describe('List component', () => {
    const props = {
        id: 1,
        name: 'name',
        itemLocation: {address1: '', city: '', state: '', zip_code: ''},
        phone: 'phne',
        price: 'price',
        img: 'link',
        website: 'link',
        rating: 1,
    }
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDom.render(<BR><ListItem {...props} /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const item = renderer.create(<BR><ListItem {...props} /></BR>);
        expect(item.toJSON()).toMatchSnapshot();
    })
})