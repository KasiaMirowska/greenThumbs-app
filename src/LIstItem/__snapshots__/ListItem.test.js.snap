// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders UI as expected 1`] = `
<div
  className="list-item"
>
  <li>
    <h2>
      name
    </h2>
    <p>
      
    </p>
    <p>
      
    </p>
    <p>
      
    </p>
    <p>
      
    </p>
    <p>
      phne
    </p>
    <p>
      price
    </p>
    <img
      src="link"
    />
    <a
      href="link"
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2>
        Visit
      </h2>
    </a>
    1
    <a
      href="/login"
      onClick={[Function]}
    >
      <button>
        Create ThumbsUP review
      </button>
    </a>
  </li>
</div>
`;
